server/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -347,6 +347,25 @@ public void testWeirdScriptFields() throws Exception {
         }
     }
 
+    // Test SearchHit "equals" fix for SearchHit with null "fields"
+    // The parsed copy of SearchHits should be equal to the original one.
+    public void testSearchHitEqualsWithNullFields() throws IOException {
```

Comment:
Instead of adding a whole new test method for this edge case, can you add a few lines to the `createTestItem` method in this test where we create random SearchHit instances for all kinds of serialization testing? Currently the only options for "fields" (L74) are either an empty map or one filled with random field names. I think we can simply add a block after that to occasionally also use "null" values to check they are serialized/deserialized correctly. This would also make it available for more equals/hashCode testing and wire serialization tests.
Theres is a useful `rarely()` method in LuceneTestCase thats accessible from tests extending ESTestCase thats true in about 5% of the times its called (roughly), I think that would be enough for testing.

Reply:
Sounds good, will update the PR with suggestion in a bit.